lib/manageiq/messaging/client.rb

```diff
@@ -90,37 +107,56 @@ def subscribe_messages(options, &block)
       # Background job assumes each job is not re-triable. It will ack as soon as a request
       # is received
       def subscribe_background_job(options)
-        BackgroundJob.subscribe(self, options)
+        assert_options(options, [:service])
+
+        subscribe_background_job_impl(options)
       end
 
       # Publish a message as a topic. All subscribers will receive a copy of the message.
       # @param options [Hash] the message attributes. Expected keys are:
       #   :service   (service is used to determine the topic address)
       #   :event     (event name)
       #   :payload   (user defined structure that describes the event)
-      #   :sender    (optional, type of the publisher)
-      #   :sender_id (optional, identity of the publisher)
+      #   :sender    (optional, identify the sender)
       #   <other queue options TBA>
       #
       def publish_topic(options)
-        Topic.publish(self, options)
+        assert_options(options, [:event, :service])
+
+        publish_topic_impl(options)
       end
 
       # Subscribe to receive topic type messages.
       # @param options [Hash] attributes to configure how to receive messages
-      #   :service    (service is used to determine the topic address)
-      #   :persist_id (optional, client needs to be have client_id to use this feature)
+      #   :service     (service is used to determine the topic address)
+      #   :persist_ref (optional, client needs to be have client_ref to use this feature)
       #
       # Persisted event: In order to consume events missed during the period when the client is
-      # offline, the subscriber needs to be reconnect always with the same client_id and persist_id
+      # offline, the subscriber needs to be reconnect always with the same client_ref and persist_ref
       #
       # A callback {|sender, event, payload| block } needs to be provided to consume the topic
       #
       def subscribe_topic(options, &block)
         raise "A block is required" unless block_given?
+        assert_options(options, [:service])
 
-        Topic.subscribe(self, options, &block)
+        subscribe_topic_impl(options, &block)
+      end
+
+      Struct.new("ManageIQ_Messaging_ReceivedMessage", :sender, :message, :payload, :ack_ref)
```

Comment:
@Fryguy I initially started with `Struct::Message`. Not sure it might conflict with a class from other gem because it looks like a popular name. Any suggestion?